src/main/java/frc/trigon/robot/subsystems/climber/ClimberVisualization.java

```diff
@@ -0,0 +1,163 @@
+
+        this.currentState = ClimberConstants.ClimberState.RESTING;
+    }
+
+    /**
+     * Updates the current climber arm angle, string angle, and string length, then logs the mechanism.
+     *
+     * @param targetState              the current climber state used to determine the second joint pose
+     * @param currentPositionRotations the current position of the climber
+     * @param targetPositionRotations  the target position of the climber
+     */
+    public void update(ClimberConstants.ClimberState targetState, double currentPositionRotations, double targetPositionRotations) {
+        this.currentState = targetState;
+
+        final Rotation2d currentFirstJointPitch = calculateClimberFirstJointPitch(calculateStringLengthMeters(currentPositionRotations));
+        final Rotation2d targetFirstJointPitch = calculateClimberFirstJointPitch(calculateStringLengthMeters(targetPositionRotations));
+        final double currentStringLength = calculateStringLengthMeters(currentPositionRotations);
+        final double targetStringLength = calculateStringLengthMeters(targetPositionRotations);
+        final Rotation2d currentStringAngle = calculateStringAngle(currentFirstJointPitch.minus(ClimberConstants.FIRST_JOINT_ANGLE_ADDITION), currentStringLength);
+        final Rotation2d targetStringAngle = calculateStringAngle(targetFirstJointPitch.minus(ClimberConstants.FIRST_JOINT_ANGLE_ADDITION), targetStringLength);
+
+        setCurrentMechanism2dState(currentFirstJointPitch, currentStringAngle, currentStringLength);
+        setTargetMechanism2dState(targetFirstJointPitch, targetStringAngle, targetStringLength);
+        log(currentFirstJointPitch);
+    }
+
+    private void setCurrentMechanism2dState(Rotation2d currentFistJointPitch, Rotation2d currentStringAngle, double currentStringLength) {
+        currentFirstJointLigament.setAngle(flipAngle(currentFistJointPitch));
+        currentStringPositionLigament.setAngle(flipAngle(currentStringAngle));
+        currentStringPositionLigament.setLength(currentStringLength);
+
+    }
+
+    /**
+     * Sets the target state of the mechanism, but doesn't log the mechanism.
+     *
+     * @param targetFirstJointPitch the target pitch of the first joint
+     * @param targetStringAngle     the string angle of the string
+     * @param targetStringLength    the target string length of the climber
+     */
+    private void setTargetMechanism2dState(Rotation2d targetFirstJointPitch, Rotation2d targetStringAngle, double targetStringLength) {
+        targetFirstJointLigament.setAngle(flipAngle(targetFirstJointPitch));
+        targetStringPositionLigament.setAngle(flipAngle(targetStringAngle));
+        targetStringPositionLigament.setLength(targetStringLength);
+    }
+
+    /**
+     * Logs the mechanism, and the poses.
```

Comment:
Explain more about the 3d poses (mention ascope)
Make sure to say "Logs the 2d mechanism"